poly/src/domain/mod.rs

```diff
@@ -81,24 +81,47 @@ pub trait EvaluationDomain<F: FftField>:
     /// Compute a IFFT, modifying the vector in place.
     fn ifft_in_place<T: DomainCoeff<F>>(&self, evals: &mut Vec<T>);
 
-    /// Multiply the `i`-th element of `coeffs` with the `i`-th power of `g`.
+    /// Multiply the `i`-th element of `coeffs` with `g^i`.
     #[cfg(not(feature = "parallel"))]
     fn distribute_powers<T: DomainCoeff<F>>(coeffs: &mut [T], g: F) {
-        let mut pow = F::one();
-        coeffs.iter_mut().for_each(|c| {
-            *c *= pow;
+        Self::distribute_powers_and_mul_by_const(coeffs, g, F::one());
+    }
+
+    /// Multiply the `i`-th element of `coeffs` with `g^i`.
+    #[cfg(feature = "parallel")]
+    fn distribute_powers<T: DomainCoeff<F>>(coeffs: &mut [T], g: F) {
+        Self::distribute_powers_and_mul_by_const(coeffs, g, F::one());
+    }
```

Comment:
There's no difference between the parallel and non-parallel versions

Reply:
Oh good point, let me remove the duplication